src/datasets/infra/repositories/transformers/datasetTransformers.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  Dataset,
+  DatasetVersionState,
+  DatasetMetadataFields,
+  DatasetMetadataBlock,
+  DatasetMetadataSubField,
+  DatasetMetadataFieldValue,
+} from '../../../domain/models/Dataset';
+import { AxiosResponse } from 'axios';
+import { NodeHtmlMarkdown } from 'node-html-markdown';
+
+export const transformVersionResponseToDataset = (response: AxiosResponse): Dataset => {
+  const versionPayload = response.data;
+  return transformVersionPayloadToDataset(versionPayload);
+};
+
+export const transformLatestVersionResponseToDataset = (response: AxiosResponse): Dataset => {
+  const versionPayload = response.data.latestVersion;
+  return transformVersionPayloadToDataset(versionPayload);
+};
+
+const transformVersionPayloadToDataset = (versionPayload: any): Dataset => {
+  return {
+    id: versionPayload.datasetId,
```

Comment:
I'm getting an error with the getDataset use case because datasetId is undefined
```suggestion
    id: versionPayload.id,
```

Error message:

```
There was an error getting the dataset
Error: undefined is not an object (evaluating 'versionPayload.datasetId')
(función anónima) — getDatasetById.ts:9
```

This is the payload:

```
{
    "status": "OK",
    "data": {
        "id": 14,
        "identifier": "FK2/3DYXBP",
        "persistentUrl": "https://doi.org/10.5072/FK2/3DYXBP",
        "protocol": "doi",
        "authority": "10.5072",
        "publisher": "Root",
        "publicationDate": "2023-05-23",
        "storageIdentifier": "local://10.5072/FK2/3DYXBP",
        "latestVersion": {
            "id": 4,
            "datasetId": 14,
            "datasetPersistentId": "doi:10.5072/FK2/3DYXBP",
            "storageIdentifier": "local://10.5072/FK2/3DYXBP",
            "versionNumber": 1,
            "versionMinorNumber": 0,
            "versionState": "RELEASED",
            "distributionDate": "2013",
            "productionDate": "2006",
            "UNF": "UNF:6:Xj34+pbwAcbJ2sZi2eqtSg==",
            "lastUpdateTime": "2023-05-23T11:31:10Z",
            "releaseTime": "2023-05-23T11:31:10Z",
            "createTime": "2023-05-23T11:31:02Z",
            "termsOfUse": "All data uploaded to this Dataverse are licensed under the CC0 licence. Please follow this link for further information: http://creativecommons.org/publicdomain/zero/1.0/",
            "fileAccessRequest": true,
            "metadataBlocks": {
                "citation": {
                    "displayName": "Citation Metadata",
                    "name": "citation",
                    "fields": [
                        {
                            "typeName": "title",
                            "multiple": false,
                            "typeClass": "primitive",
                            "value": "Data from the BAFACALO project: The Brazilian Intelligence Battery based on two state-of-the-art models: Carroll's model and the CHC model."
                        },
                        {
                            "typeName": "author",
                            "multiple": true,
                            "typeClass": "compound",
                            "value": [
                                {
                                    "authorName": {
                                        "typeName": "authorName",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "Golino, Hudson"
                                    },
                                    "authorAffiliation": {
                                        "typeName": "authorAffiliation",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "Universidade Federal de Minas Gerais"
                                    }
                                },
                                {
                                    "authorName": {
                                        "typeName": "authorName",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "Gomes, Cristiano"
                                    },
                                    "authorAffiliation": {
                                        "typeName": "authorAffiliation",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "Universidade Federal de Minas Gerais"
                                    }
                                }
                            ]
                        },
                        {
                            "typeName": "datasetContact",
                            "multiple": true,
                            "typeClass": "compound",
                            "value": [
                                {
                                    "datasetContactEmail": {
                                        "typeName": "datasetContactEmail",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "jopd@mailinator.com"
                                    }
                                }
                            ]
                        },
                        {
                            "typeName": "dsDescription",
                            "multiple": true,
                            "typeClass": "compound",
                            "value": [
                                {
                                    "dsDescriptionValue": {
                                        "typeName": "dsDescriptionValue",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "The BAFACALO's dataset presents answers from 292 Brazilian high-school students from a public school to 18 cognitive tests developed to assess different hierarchical levels of the intelligence structure. Most of the participants were girls (53.40%) and their age ranged from 14 to 20 years old (Mean = 15.71, Standard Deviation = 1.15). The BAFACALO's tests were constructed by Gomes (2005, 2010) based on the Educational Testing Service's Kit of Factor-Referenced Cognitive Tests (Ekstrom,  French,  Harman\n&  Dirmen,  1976)."
                                    },
                                    "dsDescriptionDate": {
                                        "typeName": "dsDescriptionDate",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "2013"
                                    }
                                }
                            ]
                        },
                        {
                            "typeName": "subject",
                            "multiple": true,
                            "typeClass": "controlledVocabulary",
                            "value": [
                                "Social Sciences"
                            ]
                        },
                        {
                            "typeName": "keyword",
                            "multiple": true,
                            "typeClass": "compound",
                            "value": [
                                {
                                    "keywordValue": {
                                        "typeName": "keywordValue",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "Intelligence"
                                    }
                                }
                            ]
                        },
                        {
                            "typeName": "publication",
                            "multiple": true,
                            "typeClass": "compound",
                            "value": [
                                {
                                    "publicationCitation": {
                                        "typeName": "publicationCitation",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "This data has been described in the following data paper: [insert JOPD citation]"
                                    }
                                }
                            ]
                        },
                        {
                            "typeName": "productionDate",
                            "multiple": false,
                            "typeClass": "primitive",
                            "value": "2006"
                        },
                        {
                            "typeName": "distributor",
                            "multiple": true,
                            "typeClass": "compound",
                            "value": [
                                {
                                    "distributorName": {
                                        "typeName": "distributorName",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "Harvard Dataverse Network"
                                    },
                                    "distributorURL": {
                                        "typeName": "distributorURL",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "http://thedata.harvard.edu/dvn/"
                                    },
                                    "distributorLogoURL": {
                                        "typeName": "distributorLogoURL",
                                        "multiple": false,
                                        "typeClass": "primitive",
                                        "value": "http://thedata.harvard.edu/dvn/resources/images/dvnPoweredByLogo.gif"
                                    }
                                }
                            ]
                        },
                        {
                            "typeName": "distributionDate",
                            "multiple": false,
                            "typeClass": "primitive",
                            "value": "2013"
                        },
                        {
                            "typeName": "dateOfDeposit",
                            "multiple": false,
                            "typeClass": "primitive",
                            "value": "2013-11-01"
                        }
                    ]
                }
            },
            "files": [
                {
                    "description": "",
                    "label": "bafacalo_dataset_subset.tab",
                    "restricted": false,
                    "version": 3,
                    "datasetVersionId": 4,
                    "dataFile": {
                        "id": 15,
                        "persistentId": "",
                        "filename": "bafacalo_dataset_subset.tab",
                        "contentType": "text/tab-separated-values",
                        "filesize": 80638,
                        "description": "",
                        "storageIdentifier": "local://18848607b9b-26c629f364c7",
                        "originalFileFormat": "application/x-spss-sav",
                        "originalFormatLabel": "SPSS Binary",
                        "originalFileSize": 63962,
                        "originalFileName": "bafacalo_dataset_subset.sav",
                        "UNF": "UNF:6:Xj34+pbwAcbJ2sZi2eqtSg==",
                        "rootDataFileId": -1,
                        "md5": "c9e998f23153eebc2b7c40415eac2aee",
                        "checksum": {
                            "type": "MD5",
                            "value": "c9e998f23153eebc2b7c40415eac2aee"
                        },
                        "creationDate": "2023-05-23"
                    }
                }
            ]
        }
    }
}
```